lib/act_as_attachable.rb

```diff
@@ -0,0 +1,62 @@
+module ActAsAttachable
+  extend ActiveSupport::Concern
+
+  included do
+    ActiveRecord::Base.send(:extend, ModelClassMethods)
+    ActionController::Base.send(:extend, ControllerClassMethods)
+    ActionController::Base.send(:include, ControllerInstanceMethods)
+  end
+
+  module ModelClassMethods
+    # should be declared in model, then it will be able to have attachments
+    def act_as_attachable
+      has_many :attachments, as: :attachable
+
+      accepts_nested_attributes_for :attachments, allow_destroy: true,
+                                    reject_if: lambda { |params| params[:file_upload].blank? }
```

Comment:
Line is too long. [94/80]<br>Align the elements of a hash literal if they span more than one line.